test/functional/feature_anchors.py

```diff
@@ -78,6 +81,60 @@ def run_test(self):
         self.log.info("When node starts, check if anchors.dat doesn't exist anymore")
         assert not os.path.exists(node_anchors_path)
 
+        self.log.info("Ensure addrv2 support")
+        # Use proxies to catch outbound connections to networks with 256-bit addresses
+        onion_conf = Socks5Configuration()
+        onion_conf.auth = True
+        onion_conf.unauth = True
+        onion_conf.addr = ('127.0.0.1', 10000)
+        onion_conf.keep_alive = True
+        onion_proxy = Socks5Server(onion_conf)
+        onion_proxy.start()
+        self.restart_node(0, extra_args=[f"-onion={onion_conf.addr[0]}:{onion_conf.addr[1]}"])
+
+        self.log.info(f"Add 256-bit-address block-relay-only connections to node")
+        ONION_ADDR = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion:8333"
+        self.nodes[0].addconnection(ONION_ADDR, 'block-relay-only')
+
+        self.log.info("Check peer info")
+        peer_info = self.nodes[0].getpeerinfo()
+        assert_equal(len(peer_info), 1)
+        assert_equal(peer_info[0]["network"], "onion")
+        assert_equal(peer_info[0]["connection_type"], "block-relay-only")
+
+        self.log.info("Stop node 0")
+        self.stop_node(0)
+
+        self.log.info("Check for addrv2 addresses in anchors.dat")
+        caddr = CAddress()
+        caddr.net = CAddress.NET_TORV3
+        [caddr.ip, port_str] = ONION_ADDR.split(":")
+        caddr.port = int(port_str)
+        # TorV3 addrv2 serialization:
+        # time(4) | services(1) | networkID(1) | address length(1) | address(32)
+        expected_pubkey = caddr.serialize_v2()[7:39].hex()
+
+        # position of services byte of first addr in anchors.dat
+        # network magic, vector length, version, nTime
+        services_index = 4 + 1 + 4 + 4
+        data = bytes()
+        with open(node_anchors_path, "rb") as file_handler:
+            data = file_handler.read()
+            assert_equal(data[services_index], 0x00) # services == NONE
+            anchors2 = data.hex()
+            assert expected_pubkey in anchors2
+
+        with open(node_anchors_path, "wb") as file_handler:
+            # Cheat: modify service flags for this address
+            # even though we never connected to it
+            new_data = bytearray(data)[:-32]
+            new_data[services_index] = P2P_SERVICES
+            new_data_hash = hashlib.sha256(hashlib.sha256(new_data).digest()).digest()
```

Comment:
Import helper from _test_framework.messages_?

```suggestion
            new_data_hash = hash256(new_data)

```